llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1438,13 +1438,100 @@ void VPlanTransforms::addActiveLaneMask(
     HeaderMask->replaceAllUsesWith(LaneMask);
 }
 
+static VPRecipeBase *createEVLRecipe(VPValue &EVL, VPValue *HeaderMask,
+                                     VPValue *AllOneMask,
+                                     VPRecipeBase *CurRecipe,
+                                     VPTypeAnalysis TypeInfo) {
```

Comment:
`VPTypeAnalysis &TypeInfo`.
Also, if EVL is passed by ref, beter all other params pass by ref

Reply:
done